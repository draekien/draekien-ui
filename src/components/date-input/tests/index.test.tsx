/** @jsxImportSource theme-ui */

import * as React from 'react';
import {
  fireEvent,
  getAllByText,
  getByText,
  render,
} from '@testing-library/react';
import DateInput from '..';

describe('DateInput component', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <DateInput id="datepicker" selectedDate={new Date('1 Nov 2020')} />
    );

    expect(container).toMatchSnapshot();
  });

  test('onchange function is correctly called', () => {
    const fn = jest.fn();
    const date = new Date();
    const nextDateInSeconds = new Date().setDate(new Date().getDate() + 1);
    const nextDate = new Date(nextDateInSeconds).getDate();
    const { container } = render(
      <DateInput id="datepicker" selectedDate={date} onChange={fn} />
    );

    fireEvent(
      getByText(container as HTMLElement, 'date_range'),
      new MouseEvent('click', { bubbles: true, cancelable: true })
    );

    fireEvent(
      getAllByText(container as HTMLElement, `${nextDate}`)[0],
      new MouseEvent('click', { bubbles: true, cancelable: true })
    );

    expect(fn).toHaveBeenCalledTimes(1);
  });
});
