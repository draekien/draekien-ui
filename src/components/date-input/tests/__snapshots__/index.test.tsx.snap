// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput component should match snapshot 1`] = `
<div>
  <div>
    <div
      class="css-6q34h3-DateInput"
    >
      <div
        class="css-hfz0wm-InputWrapper"
      >
        <div
          class="css-1mngvxn"
        >
          <input
            class="css-1u99oui"
            id="datepicker"
            placeholder="DD / MM / YYYY"
            value="21 / 11 / 2020"
          />
          <div
            class="css-1xauu8w-TextboxTail"
          >
            <i
              class="material-icons  css-9vbooj-Icon"
            >
              date_range
            </i>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
